app/src/main/java/com/cornellappdev/score/components/UpcomingGamesCarousel.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.compose.foundation.pager.rememberPagerState
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.cornellappdev.android.score.data.models.GameCardData
+import com.cornellappdev.score.theme.poppinsFamily
+import com.cornellappdev.score.util.gameList
+import com.example.score.R
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun DotIndicator(
+    pagerState: androidx.compose.foundation.pager.PagerState,
+    totalPages: Int,
+    modifier: Modifier = Modifier,
+    dotSize: Dp = 10.dp,
+    selectedColor: Color = MaterialTheme.colorScheme.primary,
+    unselectedColor: Color = Color.Gray
+) {
+    Row(
+        modifier = modifier,
+        horizontalArrangement = Arrangement.spacedBy(4.dp)
+    ) {
+        for (i in 0 until totalPages) {
+            val color = if (i == pagerState.currentPage) selectedColor else unselectedColor
+            Canvas(
+                modifier = Modifier
+                    .size(dotSize)
+                    .padding(2.dp)
+            ) {
+                drawCircle(color = color)
+            }
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun UpcomingGamesCarousel(games: List<GameCardData>) {
+    val pagerState = rememberPagerState(pageCount = { games.size })
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = 24.dp),
+        verticalArrangement = Arrangement.spacedBy(16.dp, Alignment.Top),
+        ) {
+        Text(
+            text = "Upcoming",
+            fontFamily = poppinsFamily,
+            fontSize = 24.sp,
+            color = Color(0xFF333333),
+            fontWeight = FontWeight(600),
+            modifier = Modifier
+                .fillMaxWidth()
+        )
+
+        HorizontalPager(
+            state = pagerState,
+            modifier = Modifier.fillMaxWidth()
+        ) { page ->
+            val game = games[page]
+            UpcomingGameCard(
+                leftTeamLogo = painterResource(R.drawable.cornell_logo),
+                rightTeamLogo = painterResource(game.teamLogo),
+                team = game.team,
+                date = game.date,
+                genderIcon = painterResource(game.genderIcon),
+                sportIcon = painterResource(game.sportIcon),
+                location = game.location,
+                modifier = Modifier,
+                headerModifier = Modifier,
+                gradientColor1 = Color(0xFFE1A69F),
+                gradientColor2 = Color(0xFF011F5B)
```

Comment:
Hm, I guess you just have the colors here as placeholders? That's cool I guess, until backend specifies how we should be getting the colors.

Reply:
Yes, will edit when knowing how colors for the gradient background will be sent in.